src/core/fetchQueue.js

```diff
@@ -0,0 +1,211 @@
+(function () {
+  'use strict';
+
+  //////////////////////////////////////////////////////////////////////////////
+  /**
+   * This class implements a queue for Deferred objects.  Whenever one of the
+   * objects in the queue completes (resolved or rejected), another item in the
+   * queue is processed.  The number of concurrently processing items can be
+   * adjusted.  At this time (2015-12-29) most major browsers support 6
+   * concurrent requests from any given server, so, when using the queue for
+   * tile images, thie number of concurrent requests should be 6 * (number of
+   * subdomains serving tiles).
+   *
+   * @class
+   *
+   * @param {Object?} [options] A configuration object for the queue
+   * @param {Number} [options.size=6] The maximum number of concurrent deferred
+   *    objects.
+   * @param {Number} [options.track=600] The number of objects that are tracked
+   *    that trigger checking if any of them have been abandoned.  The fetch
+   *    queue can grow to the greater of this size and the number of items that
+   *    are still needed.  Setting this to a low number will increase
+   *    processing time, to a high number can increase memory.  Ideally, it
+   *    should reflect the number of items that are kept in memory elsewhere.
+   *    If needed is null, this is ignored.
+   * @param {function} [options.needed=null] If set, this function is passed a
+   *    Deferred object and must return a truthy value if the object is still
+   *    needed.
+   */
+  //////////////////////////////////////////////////////////////////////////////
+  geo.fetchQueue = function (options) {
+    if (!(this instanceof geo.fetchQueue)) {
+      return new geo.fetchQueue(options);
+    }
+    options = options || {};
+    this._size = options.size || 6;
+    this._track = options.track || 600;
```

Comment:
As the comments say, picking a high number wastes memory and picking a low number wastes cpu.  The number should match the size of the tile cache, which we default to either 600 or 200 depending on whether we keep lower tiles or not (see comments in the tileLayer for why).  As we default to keeping lower tiles, that means the most common default will be 600.